src/bin/pubsub_server.rs

```diff
@@ -18,66 +16,21 @@ use actix_web::{
 };
 use clap::{App as ClApp, Arg};
 use futures::future::Future;
+use pubsub::find_service::{ConnectionResponse, PubSubResponse, ServiceStatus};
 use pubsub::{ConnectionInfo, PublisherDesc};
 use std::collections::HashMap;
 use std::net::ToSocketAddrs;
 use std::option::Option;
 use std::sync::{Arc, RwLock};
 use std::time;
 
-#[derive(Serialize, Debug)]
-struct BlankResponse {}
-
-#[derive(Serialize, Debug)]
-struct ConnectionResponse {
-    publisher: PublisherDesc,
-    info: ConnectionInfo,
-}
-
-#[derive(Serialize, Debug)]
-struct PubSubResponse<'a, T>
-where
-    T: serde::Serialize + 'a,
-{
-    status: &'a str,
-    timestamp: time::SystemTime,
-    response: &'a T,
-}
-
-impl<'a> PubSubResponse<'a, BlankResponse> {
-    fn status_only(status: &'a str) -> PubSubResponse<'a, BlankResponse> {
-        PubSubResponse {
-            status: status,
-            timestamp: time::SystemTime::now(),
-            response: &BlankResponse {},
-        }
-    }
-}
-
-impl<'a, T> PubSubResponse<'a, T>
-where
-    T: serde::Serialize,
-{
-    fn new(status: &'a str, response: &'a T) -> PubSubResponse<'a, T> {
-        PubSubResponse {
-            status: status,
-            timestamp: time::SystemTime::now(),
-            response: response,
-        }
-    }
-}
-#[derive(Serialize, Debug)]
-struct ServiceStatus {
-    count: usize,
-}
-
 type State = HashMap<String, ConnectionResponse>;
 type ProtectedState = Arc<RwLock<State>>;
 
 fn handle_service_status_request(req: HttpRequest<ProtectedState>) -> HttpResponse {
     let state = req.state().read().unwrap();
     let status = ServiceStatus { count: state.len() };
-    HttpResponse::Ok().json(PubSubResponse::new("Success", &status))
+    HttpResponse::Ok().json(PubSubResponse::new(String::from("Success"), &status))
```

Comment:
Why do you need heap allocated strings here to represent success?
These might be good candidates for 'static strs.